### PR TITLE
Upgrade NPM dependencies

### PR DESCRIPTION
**Pull request by moesoha**
Main changes:
- babel
- gulp
- eslint
- prop-types (`React.PropTypes` is deprecated)

To be continued...

### CODE REVIEW

File: scripts/build/config/gulp.js [hunk 1]
```diff
@@ -33,61 +34,66 @@ export default function ({ watch, production, errorHandler }) {
...
+
+  tasks['watch'] = () => {
     isInWatchMode = true;
     gulp.watch('vj4/ui/misc/icons/*.svg', handleWatchChange('iconfont'));
     gulp.watch('vj4/ui/constant/*.js', handleWatchChange('constant'));
     gulp.watch('vj4/locale/*.yaml', handleWatchChange('locale'));
-  });
+  };
+
+  for(let key in tasks) {
```
Comment (twd2) [inline scripts/build/config/gulp.js]:
```
for (
   ^
```
?

File: scripts/build/runGulp.js [hunk 2]
```diff
@@ -1,31 +1,36 @@
...
   }
-  gulpConfig({ watch, production, errorHandler: handleError });
+  const gulpTasks = gulpConfig({ watch, production, errorHandler: handleError });
   return new Promise(resolve => {
-    gulp.on('task_start', ({ task }) => {
-      gutil.log(chalk.blue(`Starting task: %s`), chalk.reset(task));
+    let taskList = {};
+
+    gulp.on('start', ({ uid, name }) => {
+      name[0] === '<' || log(chalk.blue(`Starting task: %s`), chalk.reset(name));
```
Comment (twd2) [inline scripts/build/runGulp.js]: What's this magic?

File: vj4/ui/components/highlighter/prismjs.js [hunk 3]
```diff
@@ -1,29 +1,15 @@
 /*
 
 To add a new language to highlight:
-1. Add new import statement in this file
+~~1. Add new import statement in this file~~
```
Comment (twd2) [inline vj4/ui/components/highlighter/prismjs.js]: I think we can just remove this line.

File: vj4/ui/components/messagepad/reducers/dialogues.js [hunk 4]
```diff
@@ -59,7 +59,7 @@ export default function reducer(state = {}, action) {
         ...state,
         [id]: data,
       };
-    } else if (type === 'reply') {
+    } if (type === 'reply') {
```
Comment (twd2) [inline vj4/ui/components/messagepad/reducers/dialogues.js]: ???

File: vj4/ui/components/scratchpad/ScratchpadRecordsRowContainer.js [hunk 5]
```diff
@@ -37,7 +35,9 @@ const getRecordDetail = (data) => {
     });
     return (
       <span key={text} className={cn}>
-        {text}: {count}
+        {text}
+:
```
Comment (twd2) [inline vj4/ui/components/scratchpad/ScratchpadRecordsRowContainer.js]: Looks strange.

File: scripts/build/runGulp.js (hunk 2 above)
Comment (moesoha) [reply scripts/build/runGulp.js]: Removed. Just let `<series>` logged.

File: vj4/ui/components/messagepad/reducers/dialogues.js (hunk 4 above)
Comment (moesoha) [reply vj4/ui/components/messagepad/reducers/dialogues.js]: strange

File: vj4/ui/components/scratchpad/ScratchpadRecordsRowContainer.js (hunk 5 above)
Comment (moesoha) [reply vj4/ui/components/scratchpad/ScratchpadRecordsRowContainer.js]: Rule `react/jsx-one-expression-per-line` disabled.

File: vj4/ui/components/messagepad/reducers/dialogues.js (hunk 4 above)
Comment (moesoha) [reply vj4/ui/components/messagepad/reducers/dialogues.js]: Refers to eslint rule `no-else-return`

**Comment by masnn** [comment]
Orz

File: vj4/ui/components/messagepad/reducers/dialogues.js (hunk 4 above)
Comment (masnn) [reply vj4/ui/components/messagepad/reducers/dialogues.js]: //eslint-disable-next-line

**Pull request merged**